ext/mbstring/mbstring.c

```diff
@@ -2076,39 +2076,43 @@ PHP_FUNCTION(mb_substitute_character)
 			RETURN_LONG(MBSTRG(current_filter_illegal_substchar));
 		}
 	} else {
-		RETVAL_TRUE;
-
-		switch (Z_TYPE_P(arg1)) {
-			case IS_STRING:
-				if (strncasecmp("none", Z_STRVAL_P(arg1), Z_STRLEN_P(arg1)) == 0) {
-					MBSTRG(current_filter_illegal_mode) = MBFL_OUTPUTFILTER_ILLEGAL_MODE_NONE;
-				} else if (strncasecmp("long", Z_STRVAL_P(arg1), Z_STRLEN_P(arg1)) == 0) {
-					MBSTRG(current_filter_illegal_mode) = MBFL_OUTPUTFILTER_ILLEGAL_MODE_LONG;
-				} else if (strncasecmp("entity", Z_STRVAL_P(arg1), Z_STRLEN_P(arg1)) == 0) {
-					MBSTRG(current_filter_illegal_mode) = MBFL_OUTPUTFILTER_ILLEGAL_MODE_ENTITY;
-				} else {
-					convert_to_long_ex(arg1);
+		if (Z_TYPE_P(arg1) == IS_STRING) {
+			if (strncasecmp("none", Z_STRVAL_P(arg1), Z_STRLEN_P(arg1)) == 0) {
```

Comment:
If `arg1` is an empty string, it's length is zero, so this condition is true.

Reply:
Ah indeed, I didn't realize this has always returned true. Which is kind of surprising but so the behaviour is as intended (not sure that I agree tho but that behaviour is already present so won't change it here).